tools/chainsync-loadtest/src/network.rs

```diff
@@ -0,0 +1,330 @@
+        Scope::run(ctx, {
+            let self_ = self.clone();
+            let ch = ch.clone();
+            move |ctx, s| async move {
+                let recv = self_.chunks.get_or_insert(&ch.chunk_hash(), || Once::new());
+                // TODO: consider converting wrapping these atomic counters into sth like a Span.
+                self_.stats.chunk_start.fetch_add(1, Ordering::Relaxed);
+                s.spawn_weak(|ctx| {
+                    self_.keep_sending(&ctx, {
+                        let ppc = self_.parts_per_chunk;
+                        move |peer| NetworkRequests::PartialEncodedChunkRequest {
+                            target: AccountIdOrPeerTrackingShard {
+                                account_id: peer.peer_info.account_id,
+                                prefer_peer: true,
+                                shard_id: ch.shard_id(),
+                                only_archival: false,
+                                min_height: ch.height_included(),
+                            },
+                            request: PartialEncodedChunkRequestMsg {
+                                chunk_hash: ch.chunk_hash(),
+                                part_ords: (0..ppc).collect(),
+                                tracking_shards: Default::default(),
+                            },
+                        }
+                    })
+                });
+                let res = ctx.wrap(recv.wait()).await;
+                self_.stats.chunk_done.fetch_add(1, Ordering::Relaxed);
+                anyhow::Ok(res?)
+            }
+        })
+        .await
+    }
+
+    fn notify(&self, msg: NetworkClientMessages) {
+        self.stats.msgs_recv.fetch_add(1, Ordering::Relaxed);
+        match msg {
+            NetworkClientMessages::NetworkInfo(info) => {
+                let mut n = self.data.lock().unwrap();
+                n.info_ = Arc::new(info);
+                if n.info_.num_connected_peers < self.min_peers {
+                    info!("connected = {}/{}", n.info_.num_connected_peers, self.min_peers);
+                    return;
+                }
+                for s in n.info_futures.split_off(0) {
+                    s.send(n.info_.clone()).unwrap();
+                }
+            }
+            NetworkClientMessages::Block(block, _, _) => {
+                self.blocks.get(&block.hash().clone()).map(|p| p.set(block));
+            }
+            NetworkClientMessages::BlockHeaders(headers, _) => {
+                if let Some(h) = headers.iter().min_by_key(|h| h.height()) {
+                    let hash = h.prev_hash().clone();
+                    self.block_headers.get(&hash).map(|p| p.set(headers));
+                }
+            }
+            NetworkClientMessages::PartialEncodedChunkResponse(resp) => {
+                self.chunks.get(&resp.chunk_hash.clone()).map(|p| p.set(resp));
+            }
+            _ => {}
+        }
+    }
+}
+
+pub struct ClientActor {
```

Comment:
maybe we should call it 'FakeClientActor'? (or Mock)

Reply:
done